### ISSUE TITLE
Further extensions to paste_linter

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1108]
Branched off from #998 -- creating this issue to cover the remaining `paste`-related lints mentioned there that aren't making the 3.0.0 cut

 - [x] #884 `PasteFilePathLinter`: `paste(sep = "/")` ➡️ `file.path()`
 - [x] #884 `PasteStrrepLinter`: `paste(rep(), collapse="")` ➡️ `strrep()` #1652
 - [ ] `paste0(x, collapse = )` ➡️ `paste(x, collapse = )`

**Comment by olivroy** [comment]
I just tripped on 

```r
paste0(x, collapse = )
```

Would love to see this implemented!

**Comment by MichaelChirico** [comment]
Could you elaborate what you mean by "tripped on"? The behavior of `paste(x, collapse=)` should be identical to that of `paste0(x, collapse=)`, the advantage here is only for consistency/possibly readability, not anything to do with correctness.

**Comment by olivroy** [comment]
Oh, actually, I tripped on `paste0(sep =)`. `sep` is ignored by `paste0()` which yielded bad results. Sorry for the confusion!

**Comment by MichaelChirico** [comment]
Your use case is already covered by `paste_linter()`:

```
lintr::lint(text='paste0(a, sep="b")', linters=lintr::paste_linter())
# <text>:1:1: warning: [paste_linter] sep= is not a formal argument to paste0(); did you mean to use paste(), or collapse=?
# paste0(a, sep="b")
# ^~~~~~~~~~~~~~~~~~
```
